extensions/data-plane/data-plane-google-storage/src/main/java/org/eclipse/edc/connector/dataplane/gcp/storage/GcsDataSinkFactory.java

```diff
@@ -74,9 +71,14 @@ public DataSink createSink(DataFlowRequest request) {
         }
 
         var destination = request.getDestinationDataAddress();
-
-        var storageClient = createStorageClient(destination.getKeyName());
-
+        var tokenKeyName = destination.getKeyName();
+        var serviceAccountKeyName = destination.getStringProperty(GcsStoreSchema.SERVICE_ACCOUNT_KEY_NAME);
+        var serviceAccountValue = destination.getStringProperty(GcsStoreSchema.SERVICE_ACCOUNT_KEY_VALUE);
+        var gcpServiceAccountCredentials = new GcpServiceAccountCredentials(tokenKeyName, serviceAccountKeyName, serviceAccountValue);
```

Comment:
Does it make sense pass directly destination.getKeyName() instead of declaring the variable? (same for SourceFactory)